src/main/java/world/bentobox/bentobox/listeners/JoinLeaveListener.java

```diff
@@ -73,9 +76,57 @@ public void onPlayerJoin(final PlayerJoinEvent event) {
             if (plugin.getIWM().inWorld(user.getLocation()) && Flags.REMOVE_MOBS.isSetForWorld(user.getWorld())) {
                 plugin.getIslands().clearArea(user.getLocation());
             }
+
+            // Clear inventory if required
+            this.clearPlayersInventory(Util.getWorld(event.getPlayer().getWorld()),
+                User.getInstance(event.getPlayer()));
+        }
+    }
+
+
+    /**
+     * This event will clean players inventor
+     * @param event SwitchWorld event.
+     */
+    @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
```

Comment:
This could work. However, we need to think about the sequence of events that occurs when a player starts a new island and make sure the clearing of the inventory happens before any commands are run or items are given to the player but after any world-switching inventory plugins swap out the inventory. Hmmm.

Reply:
Oh... I did not think about it...
I need to check it.